File: 2779. Maximum Beauty of an Array After Applying Operation/Solution.py
class Solution:
    def maximumBeauty(self, nums: List[int], k: int) -> int:
        nums.sort()
        left, right, n, high = 0, 0, len(nums), 0
        while right < n:
            while left < right and nums[right] - nums[left] > 2 * k: left += 1
            while right < n and nums[right] - nums[left] <= 2 * k: right += 1
            high = max(high, right - left)
        return high

File: 2779. Maximum Beauty of an Array After Applying Operation/Solution.java
class Solution {
    public int maximumBeauty(int[] nums, int k) {
        Arrays.sort(nums);
        int left = 0, right = 0, n = nums.length, high = 0;
        while (right < n) {
            while (left < right && nums[right] - nums[left] > 2 * k) left++;
            while (right < n && nums[right] - nums[left] <= 2 * k) right++;
            high = Math.max(high, right - left);
        }
        return high;        
    }
}
